plugins/newsdownloader.koplugin/main.lua

```diff
@@ -0,0 +1,210 @@
+local InputContainer = require("ui/widget/container/inputcontainer")
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local DEBUG = require("dbg")
+local DataStorage = require("datastorage")
+local _ = require("gettext")
+local FileManager = require("apps/filemanager/filemanager")
+
+local config = require('newsConfig');
+
+
+local NewsDownloader = InputContainer:new{}
+
+
+function NewsDownloader:init()
+    self.ui.menu:registerToMainMenu(self);
+
+    local util = require("turbo.util");
+    local feedXmlFilePath = self:getFeedXmlPath();
+
+    if not util.file_exists(feedXmlFilePath) then
+      DEBUG("Creating init configuration");
+      local newsDir = self:getNewsDirPath();
+      lfs.mkdir(newsDir);
+
+      local ffiUtil = require("ffi/util");
+      local exampleFeedPath = self:getExampleFeedPath()
+      ffiUtil.copyFile(exampleFeedPath, feedXmlFilePath);
+    end
+end
+
+
+function NewsDownloader:addToMainMenu(tab_item_table)
+    table.insert(tab_item_table.plugins, {
+        text = _("Simple News(RSS/Atom) Downloader"),
+        sub_item_table = {
+            {
+                text = _("Download news"),
+                callback = function() self:loadSourcesAndProcess(); end,
+            },
+                        {
+                text = _("Go to news folder"),
+                callback = function()
+					FileManager:showFiles(self:getNewsDirPath());
+                end,
+            },
+            {
+                text = _("Remove news"),
+                callback = function()
+						self:clearNewsDir();
+                        UIManager:show(InfoMessage:new{
+                            text = _("News removed.")
+                        })
+                end,
+            },
+            {
+                text = _("Help"),
+                callback = function()
+                        UIManager:show(InfoMessage:new{
+                            text = _("Script uses config from " .. self:getNewsDirPath() .. config.FEED_FILE_NAME .. " to download feeds to " .. self:getNewsDirPath() .. " directory."),
```

Comment:
The translation won't function this way. It needs to be something a little more like this:

```lua
text = _("Script uses config from %1%2 to download feeds to %3 directory.", self:getNewsDirPath(), config.FEED_FILE_NAME, self:getNewsDirPath()),
```